jmclient/jmclient/cryptoengine.py

```diff
@@ -18,16 +18,6 @@
 BIP44_COIN_MAP = {'mainnet': 2**31, 'testnet': 2**31 + 1}
 
 
-#
-# library stuff that should be in btc/jmbitcoin
-#
-
-
-P2PKH_PRE, P2PKH_POST = b'\x76\xa9\x14', b'\x88\xac'
-P2SH_P2WPKH_PRE, P2SH_P2WPKH_POST = b'\xa9\x14', b'\x87'
-P2WPKH_PRE = b'\x00\x14'
-
-
 def _pubkey_to_script(pubkey, script_pre, script_post=b''):
```

Comment:
Agreed and done in 
I also thought about moving `detect_script_type` to jmbitcoin but decided not to bother for now, since the TYPE_ enum is specific to jmclient atm. Can be done of course.

Additional note: made a generic `is_valid_pubkey` method in secp*main; it checks as per your original `_pubkey_to_script` on valid pubkey types, but also addresses a point I mentioned above: has a flag to insist on a compressed serialization, and this gets used in the cryptoengine engines that are segwit style.